apps/wallet-mobile/src/TxHistory/AssetList/ListBalances.tsx

```diff
@@ -0,0 +1,154 @@
+type Props = Partial<ListProps> & {
+  onScroll: ListProps['onScroll']
+  refreshing: boolean
+  onRefresh: () => void
+}
+export const ListBalances = (props: Props) => {
+  const strings = useStrings()
+  const styles = useStyles()
+  const wallet = useSelectedWallet()
+  const explorers = useExplorers(wallet.network)
+  const balances = usePortfolioBalances({wallet})
+
+  const [fungibilityFilter, setFungibilityFilter] = React.useState<Exclude<keyof typeof balances, 'records'>>('all')
+  const [amounts, setAmounts] = React.useState(balances[fungibilityFilter])
+
+  const [loadedAmounts, setLoadedAmounts] = React.useState(amounts.slice(0, batchSize))
+  const [currentIndex, setCurrentIndex] = React.useState(batchSize)
+
+  const handleOnEndReached = React.useCallback(() => {
+    if (currentIndex >= amounts.length) return
+    const nextBatch = amounts.slice(currentIndex, currentIndex + batchSize)
+    setLoadedAmounts([...loadedAmounts, ...nextBatch])
+    setCurrentIndex(currentIndex + batchSize)
+  }, [amounts, currentIndex, loadedAmounts])
+
+  const [isPending, startTransition] = React.useTransition()
+
+  const {track} = useMetrics()
+  useFocusEffect(
+    React.useCallback(() => {
+      track.assetsPageViewed()
+    }, [track]),
+  )
+
+  const handleOnChangeFilter = React.useCallback(
+    (filter: Exclude<keyof typeof balances, 'records'>) =>
+      startTransition(() => {
+        setFungibilityFilter(filter)
+        setCurrentIndex(batchSize)
+        setAmounts(balances[filter])
+        setLoadedAmounts(balances[filter].slice(0, batchSize))
+      }),
+    [balances],
+  )
+
+  const chips = React.useMemo(() => {
+    const chiplist = [
+      {label: strings.all, onPress: () => handleOnChangeFilter('all'), value: 'all', disabled: isPending},
+      {
+        label: strings.tokens(balances.fts.length),
+        onPress: () => handleOnChangeFilter('fts'),
+        value: 'fts',
+        disabled: isPending,
+      },
+    ]
+    if (balances.nfts.length > 0) {
+      chiplist.push({
+        label: strings.nfts(balances.nfts.length),
+        onPress: () => handleOnChangeFilter('nfts'),
+        value: 'nfts',
+        disabled: isPending,
+      })
+    }
+    return chiplist
+  }, [balances.fts.length, balances.nfts.length, handleOnChangeFilter, isPending, strings])
+
+  return (
+    <View style={styles.assetList} testID="assetList">
+      <FilterBalancesByType selectedValue={fungibilityFilter} chips={chips} />
+
+      <FlashList
+        {...props}
+        bounces={false}
+        data={loadedAmounts}
+        renderItem={({item: amount}) => (
+          <ExplorableAmount
+            amount={amount}
+            onPress={() => Linking.openURL(explorers.cardanoscan.token(amount.info.id))}
+          />
+        )}
+        ItemSeparatorComponent={() => <Spacer height={16} />}
+        contentContainerStyle={styles.content}
+        keyExtractor={(_, index) => index.toString()}
```

Comment:
is not possible to use tokenId as index?

Reply:
is a FlashList thing, to recycle the items in the virtualized list https://shopify.github.io/flash-list/docs/usage